### PR TITLE
[INS_EXT_POSE] Kalman Filter based on external attitude measurements

### PR DESCRIPTION
**Pull request by dewagter**
Some projects (e.g. system identification) require precise attitude from external sources. This filter merges IMU with external attitude from the EXTERNAL_POSE message.

### CODE REVIEW

File: conf/modules/ins_ext_pose.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE module SYSTEM "module.dtd">
+
+<module name="ins_ext_pose" dir="ins">
+  <doc>
+    <description>
+    extended kalman filter made by robin.
```
Comment (gautierhattenberger) [inline conf/modules/ins_ext_pose.xml:6]: Batman might be frustrated not to cited here. Any additional link or info about this filter or this particular implementation ?

File: sw/airborne/modules/ins/ins_ext_pose.c [hunk 2]
```diff
@@ -0,0 +1,1246 @@
...
+#define DEBUG_PRINT(...) {}
+#endif
+
+
+/** Data for telemetry and LTP origin.
+ */
+
+
+struct InsExtPose {
+	/* Inputs */
```
Comment (gautierhattenberger) [inline sw/airborne/modules/ins/ins_ext_pose.c:52]: run indentation script

File: sw/airborne/modules/loggers/logger_file.c [hunk 3]
```diff
@@ -43,6 +43,15 @@
 #include "firmwares/fixedwing/stabilization/stabilization_adaptive.h"
 #endif
 
+// INS
+#include INS_TYPE_H
+
+// RPM
+#ifdef BOARD_BEBOP
+#include "boards/bebop/actuators.h"
+#endif
```
Comment (gautierhattenberger) [inline sw/airborne/modules/loggers/logger_file.c:52]: these headers should be included with the `generated/modules.h`

File: sw/airborne/modules/ins/ins_ext_pose.c [hunk 4]
```diff
@@ -0,0 +1,1246 @@
...
+	// out += k4
+	float_vect_add(out, k4, EKF_NUM_STATES);
+	// out *= dt/6
+	float_vect_scale(out, dt/6, EKF_NUM_STATES);
+	// out += X
+	float_vect_add(out, X, EKF_NUM_STATES);	
+}
+
+
+static inline void ekf_step(const float U[EKF_NUM_INPUTS], const float Z[EKF_NUM_OUTPUTS], const float dt)
```
Comment (gautierhattenberger) [inline sw/airborne/modules/ins/ins_ext_pose.c:791]: would it be possible to make a generic library like https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/filters/linear_kalman_filter.h ?

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]

File: sw/airborne/modules/ins/ins_ext_pose.c (hunk 2 above)
Comment (dewagter) [reply sw/airborne/modules/ins/ins_ext_pose.c]: done

File: sw/airborne/modules/ins/ins_ext_pose.c (hunk 4 above)
Comment (gautierhattenberger) [reply sw/airborne/modules/ins/ins_ext_pose.c:800]: Of course it means that you would have to pass functions for the model and the jacobian, but that is for an other day

**Review by gautierhattenberger — APPROVED** [review]

**Pull request merged**